src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -163,28 +168,37 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
     };
 
     const esNodesCompatibility$ = pollEsNodesVersion({
-      callWithInternalUser: adminClient.callAsInternalUser,
+      callWithInternalUser: this.adminClient.callAsInternalUser,
       log: this.log,
       ignoreVersionMismatch: config.ignoreVersionMismatch,
       esVersionCheckInterval: config.healthCheckDelay.asMilliseconds(),
       kibanaVersion: this.kibanaVersion,
     }).pipe(takeUntil(this.stop$), shareReplay({ refCount: true, bufferSize: 1 }));
 
+    this.createClient = (type: string, clientConfig: Partial<ElasticsearchClientConfig> = {}) => {
+      const finalConfig = merge({}, config, clientConfig);
+      return this.createClusterClient(type, finalConfig, deps.http.getAuthHeaders);
+    };
+
     return {
       legacy: { config$: clients$.pipe(map(clients => clients.config)) },
-
-      adminClient,
-      dataClient,
       esNodesCompatibility$,
```

Comment:
~shouldn't move to `start` as well?~ ok, probably after removing adminClient & dataClient